### PR TITLE
[Ceph] Add Integration Package with Pool Disk data stream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

- Added 1 data stream (Pool Disk metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/ceph directory.
- Run the following command to run tests.
elastic-package test

## Related issues

- Relates #4871
- Relates #4930 
- Relates #4963
- Relates #5011
- Relates #5128
- Relates #5152
- Relates #5174

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM!

File: packages/ceph/data_stream/pool_disk/fields/fields.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  fields:
+    - name: pool_disk
+      type: group
+      fields:
+        - name: available.bytes
+          type: long
+          description: Available bytes of the pool.
+          metric_type: gauge
+          unit: byte
+        - name: id
```
Comment (agithomas) [inline packages/ceph/data_stream/pool_disk/fields/fields.yml:12]: please change to pool_id

File: packages/ceph/data_stream/pool_disk/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          type: long
+          description: Id of the pool.
+        - name: name
```
Comment (agithomas) [inline packages/ceph/data_stream/pool_disk/fields/fields.yml:15]: please change to pool_name

**Review by agithomas — COMMENTED** [review]
`{\"stored\":2142673,\"objects\":4,\"kb_used\":6336,\"bytes_used\":6488064,\"percent_used\":8.4362458437681198e-05,\"max_avail\":25633505280}},`

can we have these fields recorded and mapped ?

File: packages/ceph/data_stream/pool_disk/fields/fields.yml (hunk 2 above)
Comment (harnish-elastic) [reply packages/ceph/data_stream/pool_disk/fields/fields.yml:15]: In this data stream, we have pool_disk as parent field. So the current output will be like `ceph.pool_disk.name`. Based on the suggested scenario, we will have `ceph.pool_disk.pool_name`. So I think this change would make pool redundant. WDYT?

File: packages/ceph/data_stream/pool_disk/fields/fields.yml (hunk 1 above)
Comment (harnish-elastic) [reply packages/ceph/data_stream/pool_disk/fields/fields.yml:12]: In this data stream, we have `pool_disk` as parent field. So the current output will be like `ceph.pool_disk.id`. Based on the suggested scenario, we will have `ceph.pool_disk.pool_id`. So I think this change would make pool redundant. WDYT?

Comment (agithomas) [reply packages/ceph/data_stream/pool_disk/fields/fields.yml:12]: Please keept is pool_id

File: packages/ceph/data_stream/pool_disk/fields/fields.yml (hunk 2 above)
Comment (agithomas) [reply packages/ceph/data_stream/pool_disk/fields/fields.yml:15]: Please keep it as pool_name

File: packages/ceph/data_stream/pool_disk/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          type: keyword
+          description: Name of the pool.
+        - name: object.count
+          type: long
+          description: Number of objects of the pool.
+          metric_type: gauge
+        - name: used.bytes
+          type: long
+          description: Used bytes of the pool.
+          metric_type: gauge
```
Comment (agithomas) [inline packages/ceph/data_stream/pool_disk/fields/fields.yml:25]: Should this be a counter or a gauge?

Comment (harnish-elastic) [reply packages/ceph/data_stream/pool_disk/fields/fields.yml:25]: In pool if the user override/delete the data, the value can be fluctuate. We have performed the same scenario. Hence it's `gauge`.

**Review by agithomas — APPROVED** [review]
LGTM

**Pull request merged**